### PR TITLE
Add case: synthetic hurricane Sandy

### PR DESCRIPTION
**Pull request by mark-petersen**
Originally here:
1km Delaware Bay hurricane case with synthetic winds and vegetation https://github.com/MPAS-Dev/MPAS-Model/pull/724

This sets up a 1km resolution Delaware Bay hurricane run in COMPASS. It uses a parametric wind model based off of track information and the formula to calculate manning's n bottom drag from vegetation properties.

### CODE REVIEW

**Comment by xylar** [comment]
You can ignore PEP8 for merges to `legacy`.

**Comment by xylar** [comment]
@sbrus89, I think this just needs to be reviewed and can be merged (with an update in the `MPAS-Model/ocean/develop` submodule).  Do you have time to test?

**Comment by sbrus89** [comment]
@xylar, thanks for reminding me. Yes, I can test this week.

**Comment by sbrus89** [comment]
@xylar, I'm working on testing today. I'll try to approve by Monday.

**Comment by xylar** [comment]
Okay, great.  Mostly just trying to tie up loose ends with legacy COMPASS.

**Review by sbrus89 — COMMENTED** [review]
@xylar, `vegetation_tif_to_mpas.py` has a dependency on the `gdal` module which doesn't appear to be a part of the `compass_0.1.12` conda package. Was this added in a more recent version?

**Comment by xylar** [comment]
No I wasn't aware of that need.  I can make a `compass_0.2.0` that includes `gdal` but I don't recall being told that this was needed before.  Perhaps it was sneaking in via another dependency?

File: ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	<add_link source_path="script_configuration_dir" source="scripts/spinup_time_varying_forcing.py" dest="spinup_time_varying_forcing.py"/>
+	<add_link source_path="script_configuration_dir" source="hurricane_wind_pressure/main.py" dest="create_time_varying_forcing.py"/>
+	<add_link source_path="script_configuration_dir" source="synthetic_hurricanes/SANDY_017771.json" dest="SANDY_017771.json"/>
+	<add_link source_path="script_configuration_dir" source="synthetic_hurricanes/SANDY_017771_hurricane_inputs.txt" dest="hurricane_inputs.txt"/>
+	<add_link source_path="initial_condition_database" source="Chesapeake_and_Delaware_Bay_LambMap_DietrichN.tif" dest="manning.tif"/>
+	<add_link source_path="initial_condition_database" source="DelawareSM.tif" dest="vegetation_mask.tif"/>
+	<add_link source_path="initial_condition_database" source="Delaware.Size.2000-08.tif" dest="vegetation_size.tif"/>
+	<add_link source_path="initial_condition_database" source="Delaware.Height.2000-08.tif" dest="vegetaton_height.tif"/>
+	<add_link source_path="initial_condition_database" source="Delaware.Density.2000-08.tif" dest="vegetation_density.tif"/>
```
Comment (xylar) [inline ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml:18]: @sbrus89 and @caozd999, these all need to be put on the LCRC server (Anvil) and there needs to be commands for downloading them if they're missing from the local database.  See: https://github.com/MPAS-Dev/compass/blob/4cee6c61ab7cb33eddab419ffbcdc99f593d0650/ocean/global_ocean/config_files/config_initial_state.xml#L7-L25

**Comment by xylar** [comment]
`import gdal` needs to be changed to `from osgeo import gdal`.  The import to `osr` is not used and should be removed.

File: ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	<add_link source_path="script_configuration_dir" source="sandy_stations/USGS_stations.txt" dest="USGS_stations.txt"/>
+	<add_link source_path="script_configuration_dir" source="scripts/spinup_time_varying_forcing.py" dest="spinup_time_varying_forcing.py"/>
+	<add_link source_path="script_configuration_dir" source="hurricane_wind_pressure/main.py" dest="create_time_varying_forcing.py"/>
+	<add_link source_path="script_configuration_dir" source="synthetic_hurricanes/SANDY_017771.json" dest="SANDY_017771.json"/>
+	<add_link source_path="script_configuration_dir" source="synthetic_hurricanes/SANDY_017771_hurricane_inputs.txt" dest="hurricane_inputs.txt"/>
+	<add_link source_path="initial_condition_database" source="Chesapeake_and_Delaware_Bay_LambMap_DietrichN.tif" dest="manning.tif"/>
+	<add_link source_path="initial_condition_database" source="DelawareSM.tif" dest="vegetation_mask.tif"/>
+	<add_link source_path="initial_condition_database" source="Delaware.Size.2000-08.tif" dest="vegetation_size.tif"/>
+	<add_link source_path="initial_condition_database" source="Delaware.Height.2000-08.tif" dest="vegetaton_height.tif"/>
```
Comment (xylar) [inline ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml:17]:
```suggestion
	<add_link source_path="initial_condition_database" source="Delaware.Height.2000-08.tif" dest="vegetation_height.tif"/>
```

File: ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml (hunk 1 above)
Comment (xylar) [reply ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml:18]: I went ahead and added the files to Anvil.

```suggestion
 <get_file dest_path="initial_condition_database" file_name="Chesapeake_and_Delaware_Bay_LambMap_DietrichN.tif"> 
 	<mirror protocol="wget" url="https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/initial_condition_database/"/> 
 </get_file>

 <get_file dest_path="initial_condition_database" file_name="DelawareSM.tif"> 
 	<mirror protocol="wget" url="https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/initial_condition_database/"/> 
 </get_file>

 <get_file dest_path="initial_condition_database" file_name="Delaware.Size.2000-08.tif"> 
 	<mirror protocol="wget" url="https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/initial_condition_database/"/> 
 </get_file>

 <get_file dest_path="initial_condition_database" file_name="Delaware.Height.2000-08.tif"> 
 	<mirror protocol="wget" url="https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/initial_condition_database/"/> 
 </get_file>

 <get_file dest_path="initial_condition_database" file_name="Delaware.Density.2000-08.tif"> 
 	<mirror protocol="wget" url="https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/initial_condition_database/"/> 
 </get_file>

 <get_file dest_path="initial_condition_database" file_name="Chesapeake_and_Delaware_Bay_LambMap_DietrichN.tif"> 
 	<mirror protocol="wget" url="https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/initial_condition_database/"/> 
 </get_file>

	<add_link source_path="initial_condition_database" source="Chesapeake_and_Delaware_Bay_LambMap_DietrichN.tif" dest="manning.tif"/>
	<add_link source_path="initial_condition_database" source="DelawareSM.tif" dest="vegetation_mask.tif"/>
	<add_link source_path="initial_condition_database" source="Delaware.Size.2000-08.tif" dest="vegetation_size.tif"/>
	<add_link source_path="initial_condition_database" source="Delaware.Height.2000-08.tif" dest="vegetation_height.tif"/>
	<add_link source_path="initial_condition_database" source="Delaware.Density.2000-08.tif" dest="vegetation_density.tif"/>
```

**Comment by xylar** [comment]
@mark-petersen, if you have not already done so, could you make @sbrus89 a collaborator on your fork so he can make these changes and push them to this branch?

**Comment by xylar** [comment]
@sbrus89, I have updated the compass environment on Badger/Grizzly, Anvil/Chrysalis and Cori to 0.2.0, including `gdal`.  I verified on my laptop that I could run the test case through the `init` step with the changes I suggest above.

**Comment by mark-petersen** [comment]
Invited @sbrus89 as collaborator.

File: ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml (hunk 2 above)
Comment (sbrus89) [reply ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml]: Thanks for catching this!

File: ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml (hunk 1 above)
Comment (sbrus89) [reply ocean/hurricane/USDEQU60at15cr5rr1WD_veg/synthetic_sandy/config_init.xml]: Good point, @xylar. Thank you for putting the files on Anvil.

**Comment by caozd999** [comment]
@sbrus89 Thanks for taking care of these issues.

**Review by sbrus89 — APPROVED** [review]
After the fixes @xylar suggested, I tested that the `spinup` and `forward_RK4` steps run successfully on Grizzly.

**Comment by xylar** [comment]
Great!  I'm going to go ahead and merge, then.  Thanks for testing @sbrus89!

**Pull request merged**